### DISCUSSION TITLE
Establishing connectivity to G7 - possible issue and how to avoid it

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#3813] [Show and tell]
I use xDrip as a collector from G7.  
I have had cases where there was trouble establishing connectivity to a sensor for the first time.  

G6 would transmit only once every 5 minutes.  xDrip would wake up 5 minutes after the pair request approval.  This can be wrong with G7.  Before there is connectivity, G7 transmits once a minute and this behavior is referred to as "rapid reconnect".  G7 still performs only one measurement per 5 minutes.  When in rapid reconnect, it transmits the same reading 5 times.  That's how it can transmit once a minute.

After the first connection is accomplished with a pair request approval, G7 goes back to once every 5 minutes transmission.  But, the once every 5 minute interval is always on the same time grid.  
This means that the second transmission may be 1 minute or 2 minutes or 3 minutes or 4 minutes or 5 minutes after the first.  

After the pair request, I stay on the status page and make sure that xDrip continues to scan.  If it stops, I go to classic status page and tap on restart collector.  
This guarantees that xDrip will not miss the second transmission regardless of when it happens.  
After the second connection, the “Hunting Transmitter” message will disappear, and you can leave the status page.  
I have not experienced the disconnect issue any longer since I have been doing this.
I have updated the guide accordingly: [Link](https://navid200.github.io/xDrip/docs/Dexcom/G7.html)

We may automate this in xDrip so that there would be no need for manual intervention.  I still need to study the xDrip code and run more tests.  That is my plan.  
But, in the meantime, you can be cautious of this behavior and remedy the problem as explained if it occurs.

**Comment by mrwillem** [comment]
I had a problem connecting to a G7 three days ago.
I still don't know what the problem was. I had this occassionally in the past for about 12 hours or so.
However, this time the sensor refused to connect. I was also trying with a second and a third sensor but that didn't work
It was always saying Mismatch wait.
In the meantime I switched back to finger measuring with my old device.
Yesterday I noticed, that the sensor was not detectable by bluetooth scans any longer.
So I had the idea to seat it in the inserter device with the magnet again, hoping it will reboot by this.
The sensor wanted to have it's 20 minute warmup again and started as if it was new.
I still don't know what made the sensors not pairing 3 days ago.

**Reply by Navid200** [reply]
The magnet issue has been explained here: https://navid200.github.io/xDrip/docs/Dexcom/G7_Troubleshooting.html

**Reply by mrwillem** [reply]
OK. I did not read this. But what I didn't know is that the sensor will shut-down after a certain time and that you can restart it then with this trick. It was detectable at least for one and a half day. Allthough it did not pair to xdrip.

**Reply by Navid200** [reply]
This is a malfunction.  It is not a feature.  The session should start as soon as the sensor is inserted.  If not, the embedded counter effect for the impact of the immune system on the sensor behavior will not do its job.
Dexcom must be aware of the problem and I am almost certain they will fix it.

**Reply by Navid200** [reply]
This has nothing to do with xDrip.

**Reply by mrwillem** [reply]
> This has nothing to do with xDrip.

😅 I did not want to file a bug report. For me it was clear that it is not xdrip's fault.
I just wanted to share that if the sensor was detectable by bluetooth after insertion and it is not longer after a certain time, it might be resetted with the magnet so it is detectable again.
I am still wondering why the connection of four sensors did not work that night but I think I will never find out.
